In [1]:
pip install requests beautifulsoup4 lxml


Note: you may need to restart the kernel to use updated packages.


In [9]:
import requests

url = "https://www.visitmaryland.org/"

headers = {
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120 Safari/537.36"
}

response = requests.get(url, headers=headers)

print("HTTP Status Code:", response.status_code)


HTTP Status Code: 403


The request returned HTTP status code 403 (Forbidden).
The Website is protected by cloudflare and blocks automated Http request, in our case the python libraries make automated requests.
Even adding browser like headers wont work, because the server requires browser cookies to get accepted which the request cannot provide.


In [10]:
from bs4 import BeautifulSoup

soup = BeautifulSoup(response.text, "lxml")

for element in soup(["script", "style"]):
    element.decompose()

# extract visible text
text = soup.get_text()

# clean text
lines = [line.strip() for line in text.splitlines()]
clean_text = "\n".join(line for line in lines if line)

print(clean_text[:3000])   


Just a moment...Enable JavaScript and cookies to continue


In [11]:
wiki_url = "https://en.wikipedia.org/wiki/Natural_language_processing"

wiki_response = requests.get(wiki_url, headers=headers)

wiki_soup = BeautifulSoup(wiki_response.text, "lxml")

headings = wiki_soup.find_all(["h1","h2","h3"])

for h in headings:
    print(h.get_text(strip=True))


Contents
Natural language processing
History
Symbolic NLP (1950s – early 1990s)
Statistical NLP (1990s–present)
Approaches: Symbolic, statistical, neural networks
Statistical approach
Neural networks
Common NLP tasks
Text and speech processing
Morphological analysis
Syntactic analysis
Lexical semantics (of individual words in context)
Relational semantics (semantics of individual sentences)
Discourse (semantics beyond individual sentences)
Higher-level NLP applications
General tendencies and (possible) future directions
Cognition
See also
References
Further reading
External links


In [12]:
links = wiki_soup.find_all("a")

base = "https://en.wikipedia.org"

for link in links:
    href = link.get("href")
    if href and href.startswith("/wiki/"):
        print(base + href)


https://en.wikipedia.org/wiki/Main_Page
https://en.wikipedia.org/wiki/Wikipedia:Contents
https://en.wikipedia.org/wiki/Portal:Current_events
https://en.wikipedia.org/wiki/Special:Random
https://en.wikipedia.org/wiki/Wikipedia:About
https://en.wikipedia.org/wiki/Help:Contents
https://en.wikipedia.org/wiki/Help:Introduction
https://en.wikipedia.org/wiki/Wikipedia:Community_portal
https://en.wikipedia.org/wiki/Special:RecentChanges
https://en.wikipedia.org/wiki/Wikipedia:File_upload_wizard
https://en.wikipedia.org/wiki/Special:SpecialPages
https://en.wikipedia.org/wiki/Main_Page
https://en.wikipedia.org/wiki/Special:Search
https://en.wikipedia.org/wiki/Natural_language_processing
https://en.wikipedia.org/wiki/Talk:Natural_language_processing
https://en.wikipedia.org/wiki/Natural_language_processing
https://en.wikipedia.org/wiki/Natural_language_processing
https://en.wikipedia.org/wiki/Special:WhatLinksHere/Natural_language_processing
https://en.wikipedia.org/wiki/Special:RecentChangesLink

In [13]:
first_paragraph = wiki_soup.find("p")
paragraph_text = first_paragraph.get_text(strip=True)

print(paragraph_text)


Natural language processing(NLP) is the processing ofnatural languageinformation by acomputer. NLP is a subfield ofcomputer scienceand is closely associated withartificial intelligence. NLP is also related toinformation retrieval,knowledge representation,computational linguistics, andlinguisticsmore broadly.[1]


In [14]:
with open("nlp_intro.txt", "w", encoding="utf-8") as f:
    f.write(paragraph_text)

print("File saved!")


File saved!
